test/facets/thirdparty/accuse.spec.ts

```diff
@@ -0,0 +1,716 @@
+import "@nomiclabs/hardhat-waffle";
+import { expect } from "chai";
+import { deployments, ethers } from "hardhat";
+import { generateSarcophagusWithArchaeologists } from "../helpers/sarcophagus";
+import time from "../../utils/time";
+import { getContracts } from "../helpers/contracts";
+import { getFreshAccount } from "../helpers/accounts";
+import { hashShare } from "../helpers/shamirSecretSharing";
+import { BigNumber } from "ethers";
+import { getSarquitoBalance } from "../helpers/sarcoToken";
+import { SarcophagusState } from "../../types";
+import { ArchaeologistData } from "../helpers/archaeologist";
+import { verifyAccusalStatusesForArchaeologists } from "../helpers/accuse";
+
+describe("accuse v2", () => {
+  // reset to directly after the diamond deployment before each test
+  beforeEach(async () => await deployments.fixture());
+
+  describe("Validates accusal parameters", function () {
+    it("Should revert on a nonexistent sarcophagus ID", async function () {
+      const accuser = await getFreshAccount();
+
+      // generate a nonexistent sarcoId
+      const name = "does not exist";
+      const sarcoId = ethers.utils.solidityKeccak256(["string"], [name]);
+
+      // run accuse on a nonexistent sarcophagus
+      const tx = (await getContracts()).thirdPartyFacet
+        .connect(accuser)
+        .accuse(sarcoId, [], accuser.address);
+
+      await expect(tx).to.be.revertedWith(
+        `SarcophagusDoesNotExist("${sarcoId}")`
+      );
+    });
+
+    it("Should revert if the current time is past the resurrectionTime", async function () {
+      // generate a sarcophagus with a resurrection time 1 week in the future
+      const resurrectionTimeSeconds =
+        (await time.latest()) + time.duration.weeks(1);
+      const { embalmer, sarcophagus } =
+        await generateSarcophagusWithArchaeologists({
+          totalShares: 5,
+          threshold: 3,
+          maximumRewrapIntervalSeconds: time.duration.weeks(4),
+          resurrectionTimeSeconds:
+            (await time.latest()) + time.duration.weeks(1),
+        });
+
+      // set the current time to one week in the future, after the sarcophagus is allowed to be resurrected
+      await time.increaseTo(resurrectionTimeSeconds);
+
+      // accuse an archaeologist of leaking a keyshare
+      const tx = (await getContracts()).thirdPartyFacet
+        .connect(embalmer)
+        .accuse(sarcophagus.sarcoId, [], embalmer.address);
+
+      await expect(tx).to.be.revertedWith(`SarcophagusIsUnwrappable()`);
+    });
+  });
+
+  describe("Supports accusal of fewer than k archaeologists", function () {
+    it("On a successful accusal of an archaeologist, should transfer the correct amount of funds to embalmer and accuser, slash the archaeologist's bond, mark the arch as accused, and emit an AccuseArchaeologist event", async function () {
```

Comment:
I'd very much rather split up these (and other similarly combined) assertions into their own `it`-test blocks -- even if it means duplicate code.

Specifically, either a `describe` or `context` with "On successful accusal of an archaeologist", within which we have a distinct `it "should do xyz"` for each expectation.

I'm sure `beforeEach` would be enough to do the prerequisite setup being done below.

It's much easier to digest that way, and easier to debug as well.

Reply:
100%, I would love to be able to set up this test set with a context and then have an individual `it()` for most of these `expect` calls. I thought about breaking these up into more `it`s and duplicating the setup, but even with utility methods it ends up being a pretty significant amount of boilerplate getting copied between cases.